components/NavigationBar/src/MDCNavigationBar.m

```diff
@@ -453,7 +453,7 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
   if (context == kKVOContextMDCNavigationBar) {
     void (^mainThreadWork)(void) = ^{
       @synchronized(self->_observedNavigationItemLock) {
-        if (object != _observedNavigationItem) {
+        if (object != self->_observedNavigationItem) {
```

Comment:
Should this be made weak to avoid keeping the the navigationBar around if it's no longer retained otherwise?